src/geode/mesh/helpers/detail/curve_merger.cpp

```diff
@@ -0,0 +1,230 @@
+                for( const auto s : Indices{ this->meshes() } )
+                {
+                    const auto& curve = this->meshes()[s].get();
+                    new_id_[s].resize( curve.nb_edges(), NO_ID );
+                    nb_edges += curve.nb_edges();
+                }
+                edges_origins_.reserve( nb_edges );
+                curve_id_.reserve( nb_edges );
+            }
+
+            std::unique_ptr< EdgedCurve< dimension > > merge()
+            {
+                create_curve_step();
+                return this->steal_mesh();
+            }
+
+            void create_curve_step()
+            {
+                this->create_points();
+                create_edges();
+                clean_curve();
+                curve_id_.clear();
+            }
+
+            index_t edge_in_merged( index_t curve, index_t edge ) const
+            {
+                OPENGEODE_ASSERT( curve < this->meshes().size(),
+                    "[EdgedCurveMerger::edge_in_merged] Wrong curve index" );
+                OPENGEODE_ASSERT( edge < this->meshes()[curve].get().nb_edges(),
+                    "[EdgedCurveMerger::edge_in_merged] Wrong curve edge "
+                    "index" );
+                return new_id_[curve][edge];
+            }
+
+            const EdgeOrigins& edge_origins( index_t edge ) const
+            {
+                OPENGEODE_ASSERT( edge < this->mesh().nb_edges(),
+                    "[EdgedCurveMerger::edge_origins] Wrong curve edge "
+                    "index" );
+                return edges_origins_[edge];
+            }
+
+        private:
+            void clean_curve()
+            {
+                bool delete_needed{ false };
+                std::vector< bool > to_delete( this->mesh().nb_edges(), false );
+                for( const auto e : Range{ this->mesh().nb_edges() } )
+                {
+                    const auto vertices = this->mesh().edge_vertices( e );
+                    if( vertices[0] == vertices[1] )
+                    {
+                        to_delete[e] = true;
+                        delete_needed = true;
+                    }
+                }
+                if( delete_needed )
+                {
+                    const auto old2new =
+                        this->builder().delete_edges( to_delete );
+                    for( const auto s : Indices{ this->meshes() } )
+                    {
+                        const auto& curve = this->meshes()[s].get();
+                        for( const auto e : Range{ curve.nb_edges() } )
+                        {
+                            const auto old = new_id_[s][e];
+                            new_id_[s][e] = old2new[old];
+                        }
+                    }
+                }
+            }
+
+            void create_edges()
+            {
+                absl::flat_hash_map< TypedVertexCycle, index_t > edges;
+                for( const auto s : Indices{ this->meshes() } )
```

Comment:
I don't like one letter variables if they're used more than one time in the following.